core/src/main/java/org/apache/calcite/avatica/util/AbstractCursor.java

```diff
@@ -1130,18 +1165,20 @@ static class DateAccessor extends ObjectAccessor {
    */
   static class TimeAccessor extends ObjectAccessor {
     private final Calendar localCalendar;
+    private final boolean fixedInstant;
```

Comment:
Remove the `fixedInstant` field and create a separate subclass. Javadoc says 'Accessor that assumes that the underlying value is a TIME' which is a lie if you make the class serve double duty for TIME and TIME_WITH_LOCAL_TIME_ZONE.

Instant and localtime are so confusingly similar we just cannot risk storing them in the same field.

Same comments apply to the TimestampAccessor; create a class TimestampLtzAccessor.